### PR TITLE
feat: add chain selector component and support for new chains

### PR DESCRIPTION
**Pull request by juanmahidalgo**
image

### CODE REVIEW

**Review by LautaroPetaccio — APPROVED** [review]

File: webapp/src/components/Modals/BuyWithCryptoModal/BuyWithCryptoModal.tsx [hunk 1]
```diff
@@ -868,7 +886,9 @@ export const BuyWithCryptoModal = (props: Props) => {
   const renderTokenBalance = useCallback(() => {
     let balance
     if (selectedToken && selectedToken.symbol === 'MANA') {
-      balance = wallet?.networks[getNetwork(selectedChain)]?.mana.toFixed(2)
+      balance = wallet?.networks[
+        (getNetwork(selectedChain) as Network.ETHEREUM) || Network.MATIC
+      ]?.mana.toFixed(2)
```
[the remark this answers is not in the document]
Comment (juanmahidalgo) [reply webapp/src/components/Modals/BuyWithCryptoModal/BuyWithCryptoModal.tsx]: done!

File: webapp/src/components/Modals/BuyWithCryptoModal/BuyWithCryptoModal.tsx [hunk 2]
```diff
@@ -840,6 +851,13 @@ export const BuyWithCryptoModal = (props: Props) => {
             : renderBuyNowButton() // else, buy button
         }
         // for any other token, it needs to be connected on the selectedChain network
+
+        // if (wallet.network !== Network.MATIC) {
+        //   return renderBuyNowButton()
+        // }
+
+        console.log('selectedChain: ', selectedChain)
+        console.log('wallet.chainId: ', wallet.chainId)
```
[the remark this answers is not in the document]
Comment (juanmahidalgo) [reply webapp/src/components/Modals/BuyWithCryptoModal/BuyWithCryptoModal.tsx]: done!

File: webapp/src/components/Modals/BuyWithCryptoModal/BuyWithCryptoModal.tsx [hunk 3]
```diff
@@ -831,7 +842,7 @@ export const BuyWithCryptoModal = (props: Props) => {
         // it's paying with MANA but connected on Ethereum
         if (
           selectedToken.symbol === 'MANA' &&
-          wallet.network === Network.ETHEREUM
+          wallet.network !== Network.MATIC
```
[the remark this answers is not in the document]
Comment (juanmahidalgo) [reply webapp/src/components/Modals/BuyWithCryptoModal/BuyWithCryptoModal.tsx]: resolved offline ✅

File: webapp/src/components/Modals/BuyWithCryptoModal/BuyWithCryptoModal.tsx [hunk 4]
```diff
@@ -826,27 +837,36 @@ export const BuyWithCryptoModal = (props: Props) => {
...
-            ? renderSwitchNetworkButton()
-            : renderBuyNowButton() // else, buy button
-        }
-        // for any other token, it needs to be connected on the selectedChain network
+      // if can't buy Get Mana and Buy With Card buttons
+      if (!canBuyItem) {
+        return renderGetMANAButton()
+      }
+
+      // for any token other than MANA, it user needs to be connected on the destiny chain
```
Comment (LautaroPetaccio) [inline webapp/src/components/Modals/BuyWithCryptoModal/BuyWithCryptoModal.tsx:847]:
```suggestion
      // for any token other than MANA, it user needs to be connected on the destination chain
```

File: webapp/src/components/Modals/BuyWithCryptoModal/BuyWithCryptoModal.tsx [hunk 5]
```diff
@@ -826,27 +837,36 @@ export const BuyWithCryptoModal = (props: Props) => {
...
+      if (selectedToken.symbol !== 'MANA') {
```
Comment (LautaroPetaccio) [inline webapp/src/components/Modals/BuyWithCryptoModal/BuyWithCryptoModal.tsx:848]: I'm not familiar with Squid, but why does a user that wants to use the Axelar bridge needs to be connected to the destination chain? Shouldn't it work anyways?

File: webapp/src/components/Modals/BuyWithCryptoModal/BuyWithCryptoModal.tsx [hunk 6]
```diff
@@ -826,27 +837,36 @@ export const BuyWithCryptoModal = (props: Props) => {
...
+      if (asset.network === Network.ETHEREUM) {
+        // if tries to buy with ETH MANA and connected to other network, should switch to ETH network to pay directly
+        return selectedToken.symbol === 'MANA' &&
+          wallet.network !== Network.ETHEREUM &&
+          selectedChain === ChainId.ETHEREUM_MAINNET
+          ? renderSwitchNetworkButton()
+          : renderBuyNowButton()
+      }
+
+      // for L2 NFTs paying with MANA, if the price is too low, render switch button, else buy now
```
Comment (LautaroPetaccio) [inline webapp/src/components/Modals/BuyWithCryptoModal/BuyWithCryptoModal.tsx:864]: Shouldn't this be: If the asset is in the MATIC network, the price is too low, the selected chain is MATIC and the selected token is MATIC (meaning that it's natively trying to buy it, but the price is too low to cover the gas), it should show the user to switch chain.

File: webapp/src/components/Modals/BuyWithCryptoModal/BuyWithCryptoModal.tsx (hunk 5 above)
Comment (juanmahidalgo) [reply webapp/src/components/Modals/BuyWithCryptoModal/BuyWithCryptoModal.tsx:848]: Ups I think I messed up the mssage, it doesn't support meta txs! so the user has to be connected on the origin chain to trigger the tx! let me update the comment

**Pull request merged**